types/sinon/sinon-tests.ts

```diff
@@ -1,317 +1,435 @@
+    const spyTwo = sinon.spy();
 
-function testThrowsStub() {
-    sinon.stub().throws(new Error('foo'));
-    sinon.stub().throwsException(new Error('foo'));
-    sinon.stub().throws('foo');
-    sinon.stub().throwsException('foo');
-}
+    spy = sinon.spy(fn);
+    spy = sinon.spy(instance, 'foo');
 
-function testSpy() {
-    const otherSpy = sinon.spy();
-    sinon.spy().calledAfter(otherSpy);
-    sinon.spy().calledBefore(otherSpy);
-    sinon.spy().calledImmediatelyAfter(otherSpy);
-    sinon.spy().calledImmediatelyBefore(otherSpy);
-}
+    let count = 0;
+    count = spy.callCount;
 
-function testFakeServer() {
-    sinon.fakeServer.create({
-        autoRespond: true,
-        autoRespondAfter: 3,
-        fakeHTTPMethods: true,
-        respondImmediately: false
-    });
+    let called = false;
+    called = spy.called;
+    called = spy.notCalled;
+    called = spy.calledOnce;
+    called = spy.calledTwice;
+    called = spy.calledThrice;
+
+    let arr: any[] = [];
+    arr = spy.thisValues;
+    arr = spy.args;
+    arr = spy.exceptions;
+    arr = spy.returnValues;
+
+    spy('a', 'b');
+    spy(1, 2);
+    spy(true);
+
+    spy.calledBefore(spyTwo);
+    spy.calledAfter(spyTwo);
+    spy.calledImmediatelyBefore(spyTwo);
+    spy.calledImmediatelyAfter(spyTwo);
+    spy.calledWithNew();
+    spy.withArgs('a', 1).calledBefore(spyTwo);
+    spy.alwaysCalledOn(instance);
+    spy.alwaysCalledWith('a', 1);
+    spy.alwaysCalledWith('a');
+    spy.alwaysCalledWithExactly('a', 1);
+    spy.alwaysCalledWithMatch('foo');
+    spy.neverCalledWith('b', 2);
+    spy.neverCalledWith('b');
+    spy.neverCalledWithMatch('foo', 'bar');
+    spy.alwaysThrew();
+    spy.alwaysThrew('foo');
+    spy.alwaysThrew(new Error('foo'));
+    spy.alwaysReturned('foo');
+    spy.invokeCallback('a', 'b');
+    spy.calledOn(instance);
+    spy.calledWith('a', 2);
+    spy.calledWithExactly('a', 2);
+    spy.calledOnceWith('a', 2);
+    spy.calledOnceWithExactly('a', 2);
+    spy.calledWithMatch('bar', 2);
+    spy.notCalledWith('a', 2);
+    spy.notCalledWithMatch('a', 2);
+    spy.returned(true);
+    spy.returned('foo');
+    spy.returned(2);
+    spy.threw();
+    spy.threw('foo');
+    spy.threw(new Error('foo'));
+    spy.callArg(1);
+    spy.callArgOn(1, instance);
+    spy.callArgOn(1, instance, 'a', 2);
+    spy.callArgWith(1, 'a', 2);
+    spy.callArgOnWith(1, instance, 'a', 2);
+    spy.yield('a', 2);
+    spy.yieldOn(instance, 'a', 2);
+    spy.yieldTo('prop', 'a', 2);
+    spy.yieldToOn('prop', instance, 'a', 2);
+
+    let call = spy.firstCall;
+    call = spy.secondCall;
+    call = spy.thirdCall;
+    call = spy.lastCall;
+    call = spy.getCall(1);
+    call = spy.getCalls()[0];
+
+    call.calledBefore(call);
+    call.calledAfter(call);
+    call.calledWithNew();
+
+    spy.resetHistory();
+    spy.restore();
 }
 
-function testStubObject() {
-    const myObj = {
-        setStatus() {},
-        json() {}
+function testStub() {
```

Comment:
There are no tests for the inherited `Spy` methods. Maybe add something like `const spy: sinon.SinonSpy = stub` to verify that they match?

Reply:
👍  we don't need to test each individual member, just that the interface is inherited, so you are correct